CMakeLists.txt

```diff
@@ -90,86 +85,53 @@ option(IREE_ENABLE_EMITC "Enables MLIR EmitC dependencies." ${IREE_BUILD_COMPILE
-    # is still early in its development.
-    list(REMOVE_ITEM IREE_TARGET_BACKENDS_TO_BUILD WebGPU)
-  endif()
-  message(STATUS "Building compiler target backends: ${IREE_TARGET_BACKENDS_TO_BUILD}")
+# Vulkan is not natively supported on Apple platforms.
+# Metal should generally be used instead, though MoltenVK may also work.
+if (APPLE)
+  set(IREE_HAL_DRIVER_VULKAN_DEFAULT OFF)
 else()
-  set(IREE_TARGET_BACKENDS_TO_BUILD "" CACHE STRING "" FORCE)
-  message(STATUS "Compiler is disabled, building no target backends")
+  set(IREE_HAL_DRIVER_VULKAN_DEFAULT ON)
 endif()
 
-# Default every IREE_TARGET_BACKEND_* to OFF
-foreach(_backend ${IREE_ALL_TARGET_BACKENDS})
-  string(TOUPPER "${_backend}" uppercase_backend)
-  set(IREE_TARGET_BACKEND_${uppercase_backend} OFF CACHE BOOL "" FORCE)
-endforeach()
-
-# Set IREE_TARGET_BACKEND_* based on configuration
-foreach(_backend ${IREE_TARGET_BACKENDS_TO_BUILD})
-  string(TOUPPER "${_backend}" uppercase_backend)
-  string(REPLACE "\"" "" uppercase_backend ${uppercase_backend})
-  set(IREE_TARGET_BACKEND_${uppercase_backend} ON CACHE BOOL "" FORCE)
-endforeach()
-
-list(APPEND CMAKE_MODULE_PATH
-  ${CMAKE_CURRENT_LIST_DIR}/build_tools/cmake/
-  ${CMAKE_CURRENT_LIST_DIR}/bindings/python/build_tools/cmake/
-)
+option(IREE_HAL_DRIVER_CUDA "Enables the CUDA runtime HAL driver" ${IREE_HAL_DRIVER_CUDA_DEFAULT})
+option(IREE_HAL_DRIVER_DYLIB "Enables the Dylib runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_DYLIB_SYNC "Enables the Dylib_Sync runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VMVX "Enables the VMVX runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VMVX_SYNC "Enables the VMVX_Sync runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VULKAN "Enables the Vulkan runtime HAL driver" ${IREE_HAL_DRIVER_VULKAN_DEFAULT})
+
+option(IREE_TARGET_BACKEND_CUDA "Enables the CUDA compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_DYLIB-LLVM-AOT "Enables the Dylib-LLVM-AOT compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_WASM-LLVM-AOT "Enables the WASM-LLVM-AOT compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_METAL-SPIRV "Enables the Metal-SPIRV compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_ROCM "Enables the ROCm compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_VULKAN-SPIRV "Enables the Vulkan-SPIRV compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_VMVX "Enables the VMVX compiler target backend" ${IREE_BUILD_COMPILER})
+# Disable WebGPU by default - it has complex deps and is under development.
+option(IREE_TARGET_BACKEND_WEBGPU "Enables the WebGPU compiler target backend" OFF)
+
+message(STATUS "IREE build runtime HAL driver 'CUDA': ${IREE_HAL_DRIVER_CUDA}")
+message(STATUS "IREE build runtime HAL driver 'Dylib': ${IREE_HAL_DRIVER_DYLIB}")
+message(STATUS "IREE build runtime HAL driver 'Dylib_Sync': ${IREE_HAL_DRIVER_DYLIB_SYNC}")
```

Comment:
Is the weird capitalization here worth it? I kinda feel like we should just use consistent casing, which can be context dependent. So in CMake always just use all caps. This would also turn the above function into:

```cmake
function(OPTION DEFAULT)
  string(REPLACE "IREE_TARGET_BACKEND" "" NAME "${OPTION}")
  option(${OPTION} "Enables the '${NAME}' compiler target backend")
  message(STATUS "IREE build compiler target backend '${NAME}': ${OPTION}")
endfunction()
```

Reply:
Okay, swapped all dashes for underscores in the CMake options and synced up the descriptions with what the compiler and runtime flags use.

Reply:
Argh, that then needs changes in https://github.com/google/iree/blob/dacb10739393a615d5444d673b646578eb703e9a/build_tools/cmake/iree_check_test.cmake#L223-L235